### PR TITLE
Add Blank BG screen to E3V2 for S1 screen compat

### PR DESCRIPTION
**Pull request by InsanityAutomation**
The screen used on the Ender3S1 does not purge the boot screen from display memory and it commonly shows through the backgrounds corrupting graphics. Adds a new blank background screen call to flush all memory prior to drawing over it. Ive tested on both the V2 and S1 displays with Jyers and Creality UIs. Enhanced and Marlin UI is probably right but i didn't test yet.

Ill have a corresponding PR to the configs for the S1 screens as well.

### CODE REVIEW

File: Marlin/src/lcd/e3v2/proui/dwin.cpp [hunk 1]
```diff
@@ -1937,6 +1938,7 @@ void DWIN_RebootScreen() {
   DWINUI::Draw_CenteredString(Color_White, 220, GET_TEXT_F(MSG_PLEASE_WAIT_REBOOT));
   DWIN_UpdateLCD();
   delay(500);
+  DWIN_JPG_ShowAndCache(3)
```
Comment (ellensp) [inline Marlin/src/lcd/e3v2/proui/dwin.cpp]: missing ;  
should be DWIN_JPG_ShowAndCache(3);

**Review by ellensp — CHANGES_REQUESTED** [review]

File: Marlin/src/lcd/e3v2/proui/dwin.cpp (hunk 1 above)
Comment (InsanityAutomation) [reply Marlin/src/lcd/e3v2/proui/dwin.cpp]: Spotted last night but said screw it ill fix it in the AM and went to sleep! lol pushed now

**Comment by BarakRL** [comment]
FYI: Since this was merged the display on my E3V2 (4.2.7, latest DWIN) is now corrupted after booting. had to revert to a nightly build prior to this to avoid the issue.

**Comment by thisiskeithb** [comment]
> FYI: Since this was merged the display on my E3V2 (4.2.7, latest DWIN) is now corrupted after booting. had to revert to a nightly build prior to this to avoid the issue.

DId you install the latest [LCD files](https://github.com/MarlinFirmware/Configurations/tree/bugfix-2.0.x/config/examples/Creality/Ender-3%20V2)?

**Comment by BarakRL** [comment]
I did. see video below:
https://streamable.com/wb7i4f

**Comment by InsanityAutomation** [comment]
I've had GitHub mess with a graphic in the past, these need very specific encodings and sizes... If flashing the display update as well doesn't resolve it,  I'll pull it back down and check again on mine and see what happens.

**Comment by BarakRL** [comment]
it's not the LCD files I believe, even after updating rolling back to the commit before this merge fixed the issue.

**Comment by InsanityAutomation** [comment]
Which v2 display setup are you using? Creality, jyers, or proui?

**Comment by InsanityAutomation** [comment]
The display update added a page that's a blank background so if the code calls for a nonexistent display, I'd expect some oddity to follow. What you're seeing is just like what I saw on the S1 display before adding the blank background page.

**Comment by BarakRL** [comment]
im using nightly builds from https://marlin.crc.id.au/ so Pro UI (?).

**Comment by InsanityAutomation** [comment]
I'll need to dig through what he's pulling when I'm on a laptop, I'm not sure if he has the latest screen files there either as the new archive is more than just the dwin_set folder listed there. I'll touch base with crc.

**Comment by thisiskeithb** [comment]
> im using nightly builds from https://marlin.crc.id.au/ so Pro UI (?).

You'll need to use Marlin from here (this repo), not forks/third-parties.

**Comment by BarakRL** [comment]
> I'll need to dig through what he's pulling when I'm on a laptop, I'm not sure if he has the latest screen files there either as the new archive is more than just the dwin_set folder listed there. I'll touch base with crc.

I've used the dwin_set from this repo as well, not just his. same thing.

**Comment by ellensp** [comment]
You said "latest DWIN"  Check the chip on back of LCD, if it is readable and says F1C100s, this uses data from "private" directory, not the DWIN_SET directory   
F1C100s

**Comment by BarakRL** [comment]
> > im using nightly builds from https://marlin.crc.id.au/ so Pro UI (?).
> 
> You'll need to use Marlin from here (this repo), not forks/third-parties.

I've pulled bugfix-2.0.x to my computer and compiled it on my machine (took me some time to config it all since there's no 4.2.7 example for the E3V2) and got the same result. 
Corrupted display on latest commit (commit 323ac9488ce0562611d0144ab8ba700d272c00c7).
Resetting to last commit before this PR (commit 7d7439c4b62e3d353bb20158c04d60defe89c694) - no issue.

More info:
I'm using `DWIN_CREALITY_LCD_ENHANCED`, with `BOARD_CREALITY_V427` and the OEM screen that came with the E3V2.
LCD files updated, using DWIN_SET (which is the right FW for my screen, I've been switching to jyers and back so I know it works).

If you want my configs let me know.

**Comment by InsanityAutomation** [comment]
The last part of the equation is the screen files themselves, did you get the screen update added here

https://github.com/MarlinFirmware/Configurations/tree/import-2.0.x/config/examples/Creality/Ender-3%20V2/LCD%20Files as this was modified to match this pull request.

As mentioned above though the enhanced or pro-UI is one that I did not physically test. The others did get tested as they are currently the more prevalent. Most likely work will be done in the near future to merge all of them together and eliminate the need for testing multiple UIs.

**Comment by BarakRL** [comment]
> The last part of the equation is the screen files themselves, did you get the screen update added here
> 
> https://github.com/MarlinFirmware/Configurations/tree/import-2.0.x/config/examples/Creality/Ender-3%20V2/LCD%20Files as this was modified to match this pull request.
> 
> As mentioned above though the enhanced or pro-UI is one that I did not physically test. The others did get tested as they are currently the more prevalent. Most likely work will be done in the near future to merge all of them together and eliminate the need for testing multiple UIs.

I actually tried pulling from both `bugfix-2.0.x` and `import-2.0.x` just to make sure (`import-2.0.x` is a few commits ahead but both contains #687 it seems). 
Alas, same behavior.

**Comment by CRCinAU** [comment]
The `DWIN_SET.zip` file in my nightlies are packaged each night from the Configurations repo. As such, it should always contain matching content for the LCD. Todays packing includes the `private` folder as well.

The builds are made with the enhanced UI, and are all based off `bugfix-2.0.x` for both Configurations and Marlin source.

I tested the latest nightly, and the DWIN_SET from the Configurations repo - and can confirm the results as per the video posted by @BarakRL

**Comment by CRCinAU** [comment]
Interestingly, reverting `DWIN_Startup()` in `Marlin/src/lcd/e3v2/proui/dwin.cpp` seems to restore correct behaviour - ie:

```
// Startup routines
void DWIN_Startup() {
  DWINUI::init();
  DWINUI::onCursorDraw = Draw_Menu_Cursor;
  DWINUI::onCursorErase = Erase_Menu_Cursor;
  DWINUI::onTitleDraw = Draw_Title;
  DWINUI::onMenuDraw = Draw_Menu;
  HMI_SetLanguage();
}
```

@InsanityAutomation - Is there some reason for adding `DWIN_JPG_ShowAndCache(3);` here? It seems to do more bad than good?

**Comment by CRCinAU** [comment]
@BarakRL - I've added a patch to restore DWIN_Startup() as per my above comment and rebuilt all firmware varients for the Ender 3 V2. Can you please grab the latest nightly to test?

They should have the datestamp `20220306`.

**Pull request merged**